backend/api/routes.py

```diff
@@ -133,3 +136,30 @@ def selectOption(name):
         return jsonify({'response': 'ok'})
     else:
         return jsonify({'response': 'error', 'message': msg})
+
+@app.route('/uploadRecipe', methods = ['POST'])
+def uploadRecipe():
+    """
+    Uploads a file to the recipes folder, file must be valid JSON.
+
+    :return:
+    object
+        response
+            One of:
+                ok
+                error
+        message
+            Only present if response is "error" and there is a message to present to the user.
+    """
+    
+    f = request.files['File']
```

Comment:
What happens if someone adds multiple file variables?  I think this would cause a runtime error when checking the minetype

Reply:
request.files is an ImmutableMultiDict where each key corresponds to an individual file, so if the files have different keys only the one with the key "File" will get processed. Looks like for whatever reason an ImmutableMultiDict can have two items with the same key and request.files['File'] grabs the first item in the request payload that had that key. I think if it just processes the one file it's fine, but I can change it so it will process all files in the dict instead if you think that would be better.

Reply:
let's leave it as is and I'll test to see what behavior happens when http treats files as an array